highfidelity/hifi
# Beta Release 23 - Includes up to Developer Release 5456

http://rawgit.com/cozza13/QAtesting/master/QAofRC.js

**Specific Tutorial tests**

**Preparation**

Run through this each time you want to test this PR to ensure that you are in a "clean" state.

 1. Clear out your settings. You can do so by deleting (back it up if you need it) `C:/Users/USERNAME/AppData/Roaming/High Fidelity/Interface.ini`
 1. Delete:
   1. `C:\Users\USERNAME\AppData\Local\High Fidelity`
   1. `C:\Users\USERNAME\AppData\Local\High Fidelity - PR8627` (You might not have this)
   1. `C:\Users\USERNAME\AppData\Roaming\High Fidelity - PR8627` (You might not have this)


**To test the tutorial with the Vive:**

 1. On first load you should be taken to a dark room with tatami mats on the ground. Shortly after you should see text that tells you to raise your hands above your head. If you don't get to this point there is something wrong.
 1. Follow the tutorial - it should be self-explanatory. The steps are:
   1. Raise hands above head
   1. Raise hands above head (again)
   1. Near grab
   1. Far grab
   1. Equip lighter and light fuse
   1. Unequip lighter
   1. Turn using controller
   1. Teleport
   1. Go to portal on the right
 1. You should not be able to use far grab, equip, or teleport before you get to those steps.
 1. You should not be able to open the menu using the button above the trackpad.
 1. You should not see your avatar or the HUD (other than the HUD button) during the tutorial.
   1. You can click the HUD, open the address bar, etc., if you know what you are doing, but that is OK.
 1. Try playing around in the tutorial, see if there is anything that breaks through normal use. This should not include turning the HUD on, using your keyboard, etc., though.
 1. Did you notice any major issues?
1. Did you hear: environmental sounds, the sound of advancing to next level, the sound of the lighter clicking and butane, the sound of the firework exploding in the fire, the sound of the firework spinner?
1. Did you see the firework particles? 


**If you are not on a Vive:**

 1. On first launch (by clicking "Go Home"):
   1. You should see a help window popup that looks like https://github.com/huffman/hifi/blob/951a36854fdf0e9fe34cc4d373c2f21110e53ee2/interface/resources/html/img/controls-help-gamepad.jpg
     1. You should be able to click on the 3 tabs (KB/Mouse, Hand controllers, Game Controller)
   1. You should be taken to `Welcome`
 1. On second launch (by clicking "Go Home") you should be taken to your home server


**Parts of the client that have changed and should be tested in particular:**
 1. teleport.js
 1. handControllerGrab.js
 1. ATP downloading (are atp downloads failing?) (you don't have to test this directly, but go to areas that would stress this)
 1. Script downloading (are script downloads acting oddly?) (you don't have to test this directly, but go to areas that would stress this)





* display location and connection status in address bar when it doesn't have focus 
* Support external GL textures with proper fencing
* Initial work towards making performance testing reproducible
* Remove all usage of oglplus
* Fix model overlay visibility and texture properties
* Add ability to parent overlays to MyAvatar for use across servers
* Add category to log lines
* Fix intermittent channel swapping
* Fix CMAKE warning about un-unit-testable CPP file in physics unit tests
* Add flag to interface to have it start the sandbox server
* Fix 16khz support and hot-unplugging audio devices
This PR fixes two unrelated audio issues on Windows, by updating qtaudio_wasapi.dll. 
* Fix ScriptEngine not properly resolving paths
* Trigger unequip at a more lenient angle and with the secondary trigger
* Add setMapping to ATP scripting interface and fix threading bug
* added a command-line tool for checking on health of a domain-server and ACs 
* Add retrying of script requests
* Remove QML threaded rendering
* Fix script paths not resolving correctly
* HMD Keyboard
* Kill agents in avatar mixer when they stop being avatars
* Update teleport.js to ignore invisible or collisionless objects and disallow teleporting to certain surfaces
* init the accumulator for script elapsed time
* Updated Tutorial content
* Fix page size usage in gpu buffers
